include/agl/shader/aglShaderCompileInfo.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <hostio/seadHostIONode.h>
+#include "agl/shader/aglShader.h"
+
+namespace sead {
+class Heap;
+}
+
+namespace agl {
+
+class ShaderCompileInfo : public sead::hostio::Node {
+    // this value is used as an index to a table of version lists
+    // on SMO 1.2.0, located at 0x7101E80B30
+    using Target = s32;
+
+    ShaderCompileInfo();
+
+    virtual ~ShaderCompileInfo();
+
+    void destroy();
+    void create(s32, s32 bufferSize, sead::Heap*);
+    void clearVariation();
+    void pushBackVariation(const char*, const char*);
+    void calcCompileSource(ShaderType, sead::BufferedSafeString*,
+                           ShaderCompileInfo::Target, bool);
```

Comment:
Is ``ShaderCompileInfo::`` necessary here? Also afaik we wouldn't need it to fit our styling guide